src/sinks/pulsar/config.rs

```diff
@@ -0,0 +1,298 @@
+use crate::sinks::util::TowerRequestConfig;
+use crate::{
+    codecs::EncodingConfig,
+    config::{AcknowledgementsConfig, GenerateConfig, Input, SinkConfig, SinkContext},
+    schema,
+    sinks::{
+        pulsar::sink::{healthcheck, PulsarSink},
+        Healthcheck, VectorSink,
+    },
+};
+use codecs::{encoding::SerializerConfig, TextSerializerConfig};
+use futures_util::FutureExt;
+use pulsar::authentication::oauth2::{OAuth2Authentication, OAuth2Params};
+use pulsar::error::AuthenticationError;
+use pulsar::{
+    compression, message::proto, Authentication, Error as PulsarError, ProducerOptions, Pulsar,
+    TokioExecutor,
+};
+use snafu::ResultExt;
+use value::Kind;
+use vector_common::sensitive_string::SensitiveString;
+use vector_config::configurable_component;
+use vector_core::config::DataType;
+
+/// Configuration for the `pulsar` sink.
+#[configurable_component(sink("pulsar"))]
+#[derive(Clone, Debug)]
+pub struct PulsarSinkConfig {
+    /// The endpoint to which the Pulsar client should connect to.
+    ///
+    /// The endpoint should specify the pulsar protocol and port.
+    #[serde(alias = "address")]
+    #[configurable(metadata(docs::examples = "pulsar://127.0.0.1:6650"))]
+    pub(crate) endpoint: String,
+
+    /// The Pulsar topic name to write events to.
+    #[configurable(metadata(docs::examples = "topic-1234"))]
+    pub(crate) topic: String,
+
+    /// The name of the producer. If not specified, the default name assigned by Pulsar will be used.
```

Comment:
Is that default something we don't know (if we did, we could make this not an `Option`), but only the Pulsar server does?

Reply:
I have zero experience with this service but I believe that it is one that the Pulsar server generates, thus should be an Option here:

```
producer_name: (optional) If a producer name is specified, the name will be used, otherwise the broker will generate a unique name. Generated producer name is guaranteed to be globally unique. Implementations are expected to let the broker generate a new producer name when the producer is initially created, then reuse it when recreating the producer after reconnections.
```

https://pulsar.apache.org/docs/2.11.x/developing-binary-protocol/#command-producer

Reply:
👍 I agree with your assessment there, it wasn't clear from our doc if it was a "default" name we could reference - but as the broker generates it `Option` seems appropriate.